icu4j/main/core/src/test/java/com/ibm/icu/dev/test/lang/UnicodeSetTest.java

```diff
@@ -3198,4 +3205,178 @@ public void TestPatternCodePointComplement() {
+        }
+    }
+
+    @Test
+    public void testIterationMethodsRanges() {
+        final UnicodeSet us = populateUnicodeSet();
+
+        // Build the reference test result
+        StringJoiner expected = newResult();
+        for (EntryRange r : us.ranges()) {
+            expected.add(rangeToString(r));
+        }
+
+        UnicodeSetIterator usi = new UnicodeSetIterator(us);
+        StringJoiner fromUnicodeSetIterator = newResult();
+        while (usi.nextRange()) {
+            if (usi.codepoint != UnicodeSetIterator.IS_STRING) {
+                fromUnicodeSetIterator.add(rangeString(usi.codepoint, usi.codepointEnd));
+            }
+        }
+        assertEquals("ranges :: UnicodeSetIterator", expected.toString(), fromUnicodeSetIterator.toString());
+
+        StringJoiner fromStream = newResult();
+        us.rangeStream().forEach(r -> fromStream.add(rangeToString(r)));
+        assertEquals("ranges :: rangeStream", expected.toString(), fromStream.toString());
+    }
+
+    @Test
+    public void testIterationMethodsStrings() {
+        final UnicodeSet us = populateUnicodeSet();
+
+        // Build the reference test result
+        StringJoiner expected = newResult();
+        for (String str : us.strings()) {
+            expected.add(stringToString(str));
+        }
+
+        UnicodeSetIterator usi = new UnicodeSetIterator(us);
+        StringJoiner fromUnicodeSetIterator = newResult();
+        while (usi.nextRange()) {
+            if (usi.codepoint == UnicodeSetIterator.IS_STRING) {
+                fromUnicodeSetIterator.add(stringToString(usi.getString()));
+            }
+        }
+        assertEquals("strings :: UnicodeSetIterator", expected.toString(), fromUnicodeSetIterator.toString());
+
+        StringJoiner fromStream = newResult();
+        us.stringStream().map(UnicodeSetTest::stringToString).forEach(fromStream::add);
+        assertEquals("strings :: stringStream", expected.toString(), fromStream.toString());
+    }
+
+    @Test
+    // Iterates on strings AND on code points (converted to strings)
+    public void testIterationMethodsCodepointsAndStrings() {
+        final UnicodeSet us = populateUnicodeSet();
+
+        // Build the reference test result
+        StringJoiner expect = newResult();
+        for (String str : us) {
+            expect.add(stringToString(str));
+        }
+
+        StringJoiner fromStream = newResult();
+        us.stream().map(UnicodeSetTest::stringToString).forEach(fromStream::add);
+
+        assertEquals("code points + strings :: stream", expect.toString(), fromStream.toString());
+    }
+
+    /*
+     * Helper methods for testing various iterations.
+     * The main goal is to make the results of any possible failures more readable
+     * by formatting code points to something like U+03A3(Σ) and wrapping strings in double quotes.
+     */
+
+    private static String codePointToString(int codepoint) {
```

Comment:
A "code point" variable should be called camelCase codePoint -- or simply cp or c.